control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -117,6 +117,11 @@ type HostedControlPlaneReconciler struct {
 	HostedAPICache  hostedapicache.HostedAPICache
 	upsert.CreateOrUpdateProvider
 	EnableCIDebugOutput bool
+	// RegistryOverrides contains the source registry string as a key and the destination registry string as value.
+	// images before being applied are scanned for the source registry string and if found the string is replaced with
+	// the destination registry string. This allows hypershift to run in non-crio environments where mirroring is not
+	// applicable.
+	RegistryOverrides map[string]string
```

Comment:
Nice to have: define a type for this and implement the flag.Value interface for it.